### PR TITLE
Unified RAM inference code

### PR DESCRIPTION
**Pull request by Miles-Ma**
This PR gives a new `inference_ram.py`, replacing the old `inference_ram.py` and `inference_ram_openset.py`.

The new `inference_ram.py` supports:
- batchified inference on all 4585 categories or specified subsets
- open-set inference on custom unseen categories
- inference on custom dataset
- adjustable threshold
- evaluation (mAP, Precision, Recall)
- logits saving

Formally released two datasets: `openimages_common_214` and `openimages_rare_200` (used in our paper).

Also updated `.gitignore`, `readme.md` and `requirements.txt` accordingly.

### CODE REVIEW

**Comment by Miles-Ma** [comment]
This PR is too large, close it for now.
I will first adjust folder structures, then merge these changes gradually.
